rfcs/libraries.md

```diff
@@ -0,0 +1,128 @@
+# Libraries
+
+## Summary
+
+We need a way to group together related code and provide easy ways to require them.
+
+## Motivation
+
+- Package management
+- Third party libraries
+- Simpler requires
+
+## Design
+
+### String Syntax
+
+Our require syntax should:
+
+- Allow a library written in Luau to be imported into the Roblox engine and 'just work'.
+- Support compile-time file resolution _where possible_ for type checking.
+
+#### Absolute
+
+Modules can be required by their absolute path from the root of the filesystem.
+
+Requiring a file called `MyModule.luau` in `C:/MyLibrary`:
+```lua
+local MyModule = require("C:/MyLibrary/MyModule")
+```
+
+Generally, absolute paths should not be used in regular code and should only appear in the alias map. Their purpose is to allow libraries to be stored in a global location (such as if a package is installed globally) without breaking if the libraries location is changed on disk.
+
+#### Relative
+
+Modules can be required relative to the requiring files location in the filesystem.
+
+If we are trying to access a file called `MyModule.luau` in `C:/MyLibrary`:
+```lua
+-- From C:/MyLibrary/SubDirectory/SubModule.luau
+local MyModule = require("../MyModule")
+ 
+-- From C:/MyOtherLibrary/MainModule.luau
+local MyModule = require("../MyLibrary/MyModule")
+```
+
+All relative paths will start with either `./`  or `../`  which denote the directory of the script or file, and parent directory respectively.
+
+#### Aliases
+
+Aliases can be used to bind an absolute or relative path to a convenient name that can be required directly. They are always prefixed with `$` to make it obvious they are not the same as a regular path.
+
+Each library has its own map which will be stored in the `.luaurc` file. Since libraries can be embedded inside of one another, any aliases which are not overriden will be inherited from the parent library.
+
+```json
+"Aliases": {
+    "Roact": "C:/LuauModules/Roact-v1.4.2"
```

Comment:
How will alias look for `DataModel`?

Reply:
The caveat of using paths like `/ReplicatedStorage/Roact` is that in networked DM modules may be not available yet and in real codebases looks something like this:
```
local ReplicatedStorage = game:GetService("ReplicatedStorage")
local Roact = require(ReplicatedStorage::WaitForChild("Roact"))
```
This translation of FS based paths to DM based paths might not work as simple substitution

Reply:
This is covered in internal documentation